src/storage/memory_store.ts

```diff
@@ -302,6 +351,11 @@ export class MemoryStore implements IStore {
   async saveTx(tx: IHistoryTx): Promise<void> {
     this.history.set(tx.tx_id, tx);
 
+    // Add transaction to the ordering list
+    // and sort it so we ensure the order
+    this.historyTs.push(getOrderingKey(tx));
+    this.historyTs.sort();
```

Comment:
Should we expect heavy CPU loads from this sorting?

Considering that the new storage scheme is aimed at heavy users, I can imagine this could have an impact on a heavily used exchange wallet, for example. I'm imagining receiving many new transactions per second on a wallet that has thousands of transactions/addresses.

Reply:
For new transactions, pushing to the end should already be in order since we are ordering by timestamp and the most recent is the higher timestamp.
This should add some overhead when loading the history but it is necessary to ensure this feature.

If this raises an issue later we could add an option to the config to disable this.